src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Runtimes/MissingSdkRuntimeDetector.cs

```diff
@@ -18,20 +18,29 @@ internal class MissingSdkRuntimeDetector : OnceInitializedOnceDisposedAsync, IPr
         private Guid _projectGuid;
 
         private readonly ConfiguredProject _project;
-        private readonly IProjectFaultHandlerService _projectFaultHandlerService;
         private readonly IMissingSetupComponentRegistrationService _missingSetupComponentRegistrationService;
+        private readonly IProjectSubscriptionService _projectSubscriptionService;
+
+        private IDisposable? _subscription;
 
         [ImportingConstructor]
         public MissingSdkRuntimeDetector(
             IMissingSetupComponentRegistrationService missingSetupComponentRegistrationService,
             ConfiguredProject configuredProject,
             IProjectThreadingService threadingService,
-            IProjectFaultHandlerService projectFaultHandlerService)
+            IProjectFaultHandlerService projectFaultHandlerService, 
+            IProjectSubscriptionService projectSubscriptionService)
             : base(threadingService.JoinableTaskContext)
         {
             _missingSetupComponentRegistrationService = missingSetupComponentRegistrationService;
             _project = configuredProject;
-            _projectFaultHandlerService = projectFaultHandlerService;
+            _projectSubscriptionService = projectSubscriptionService;
+        }
+        
+        
+        private Task OnProjectChangedAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> _)
+        {
+            return RegisterSdkNetCoreRuntimeNeededInProjectAsync(_project);
         }
```

Comment:
Consider either making this a local function in the method that subscribes to data, or moving it down in the file so that it comes after the place it's used. Generally we move lifetime methods like this down below things like load/unload methods. Or thought about another way, we introduce methods generally _after_ we call them, so that entry level methods come higher in the file.